ee/vellum_ee/workflows/display/nodes/vellum/final_output_node.py

```diff
@@ -33,6 +48,24 @@ def serialize(self, display_context: WorkflowDisplayContext, **_kwargs) -> JsonO
         )
         inferred_type = infer_vellum_variable_type(node.Outputs.value)
 
+        output_obj: Dict[str, Any] = {
+            "id": str(self._get_output_id()),
+            "name": node.Outputs.value.name,
+            "type": inferred_type,
+            "value": serialize_value(node_id, display_context, node.Outputs.value.instance),
+        }
+
+        if inferred_type == "REFERENCE":
+            # Get the Python type from the descriptor
+            python_type = node.Outputs.value.types[0] if node.Outputs.value.types else None
+            if python_type:
+                schema = _generate_json_schema_for_type(python_type)
```

Comment:
we should be able to use our existing `compile_annotation` helper instead of this new method